plugins/providers/alicloud_ram/client.go

```diff
@@ -0,0 +1,216 @@
+	reqClient, err := c.newRequestClient()
+	if err != nil {
+		return err
+	}
+
+	// TODO: find a way to add parent context to the request
+	_, err = reqClient.DetachPolicyFromRoleWithOptions(&ram.DetachPolicyFromRoleRequest{
+		PolicyName: &policyName,
+		PolicyType: &policyType,
+		RoleName:   &roleName,
+	}, &utils.RuntimeOptions{})
+	if err != nil {
+		// TODO: find the error list on SDK instead of using strings contains
+		if strings.Contains(err.Error(), "EntityNotExist.Role.Policy") {
+			return ErrPermissionNotExist
+		}
+		return err
+	}
+
+	return nil
+}
+
+func (c *aliCloudRAMClient) ListAccess(_ context.Context, _ domain.ProviderConfig, _ []*domain.Resource) (domain.MapResourceAccess, error) {
+	return nil, ErrUnimplementedMethod // TODO
+}
+
+func (c *aliCloudRAMClient) GetAllPoliciesByType(_ context.Context, policyType string, maxItems int32) ([]*ram.ListPoliciesResponseBodyPoliciesPolicy, error) {
+	reqClient, err := c.newRequestClient()
+	if err != nil {
+		return nil, err
+	}
+
+	result := make([]*ram.ListPoliciesResponseBodyPoliciesPolicy, 0)
+	var marker *string
+	for {
+		// TODO: find a way to add parent context to the request
+		resp, err := reqClient.ListPoliciesWithOptions(&ram.ListPoliciesRequest{
+			Marker:     marker,
+			MaxItems:   &maxItems,
+			PolicyType: &policyType,
+		}, &utils.RuntimeOptions{})
+		if err != nil {
+			return nil, err
+		}
+
+		result = append(result, resp.Body.Policies.Policy...)
+		if resp.Body.Marker == nil {
+			break
+		}
+		marker = resp.Body.Marker
+	}
+
+	return result, nil
+}
+
+// newRequestClient creates a new RAM client instance for each request.
+//
+// AliCloud SDK clients are not concurrency-safe due to their use of a builder pattern
+// for sending and receiving requests. To avoid race conditions, we must create a
+// new client instance for every request.
+//
+// The client uses RAM (Resource Access Management) credentials to authenticate with AliCloud.
+// By default, it uses access key credentials. If a role ARN (`ramRole`) is specified,
+// it assumes that role to generate temporary session credentials.
+func (c *aliCloudRAMClient) newRequestClient() (*ram.Client, error) {
+	// Default to access key credentials (RAM User)
+	credentialConfig := &credentials.Config{
+		Type:            bptr.FromString(aliAccountTypeAccessKey),
+		AccessKeyId:     &c.accessKeyId,
+		AccessKeySecret: &c.accessKeySecret,
+	}
+
+	// If a role to assume is specified, configure credentials to assume the role (RAM Role)
+	if c.ramRole != "" {
+		credentialConfig.Type = bptr.FromString(aliAccountTypeRamRoleARN)
+		credentialConfig.RoleArn = &c.ramRole
+		credentialConfig.RoleSessionExpiration = bptr.FromInt(aliRoleSessionExpiration)
+	}
+
+	credential, err := credentials.NewCredential(credentialConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a new credentials: %w", err)
+	}
+
+	reqClient, err := ram.NewClient(&openapi.Config{Credential: credential})
```

Comment:
Let's also specify which RegionID to use while creating the client.
We can define RegionID in ProviderConfig's Credentials.

Reply:
Ack!

Reply:
Resolved: [ce1cbfb](https://github.com/goto/guardian/pull/190/commits/ce1cbfb5780b926b5fe1de2ad011281355f8a801)